airbyte-integrations/connectors/source-aws-cloudtrail/source_aws_cloudtrail/source.py

```diff
@@ -0,0 +1,230 @@
+
+        while not pagination_complete:
+            params = self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+            response = self.send_request(**params)
+
+            next_page_token = self.next_page_token(response)
+            if not next_page_token:
+                pagination_complete = True
+
+            for record in self.parse_response(response):
+                yield record
+                self.records_left -= 1
+
+                if self.is_read_limit_reached():
+                    return iter(())
+
+        yield from []
+
+
+class IncrementalAwsCloudtrailStream(AwsCloudtrailStream, ABC):
+
+    # API does not support read in ascending order
+    # save state only once after full read
+    state_checkpoint_interval = None
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        record_time = latest_record[self.time_field]
+        return {self.cursor_field: max(record_time, current_stream_state.get(self.cursor_field, 0))}
+
+
+class ManagementEvents(IncrementalAwsCloudtrailStream):
+    primary_key = "EventId"
+
+    time_field = "EventTime"
+
+    cursor_field = "EventTime"
+
+    data_field = "Events"
+
+    day_in_seconds = 24 * 60 * 60
+
+    data_lifetime = 90  # in days
+
+    def send_request(self, **kwargs) -> Mapping[str, Any]:
+        return self.client.session.lookup_events(**kwargs)
+
+    def parse_response(self, response: dict, **kwargs) -> Iterable[Mapping]:
+        for event in response[self.data_field]:
+            # boto3 converts timestamps to datetime object
+            # we need to convert it back to timestamp to persist original API type
+            event[self.time_field] = self.datetime_to_timestamp(event[self.time_field])
+            yield event
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        """
+        Slices whole time range to more granular slices (24h slices). Latest time slice goes first to avoid data loss
+        """
+        cursor_data = stream_state.get(self.cursor_field) if stream_state else None
+        # ignores state if start_date option is higher than cursor
+        if cursor_data and cursor_data > self.start_date:
+            start_time = self.normalize_start_time(cursor_data)
+        else:
+            start_time = self.normalize_start_time(self.start_date)
+
+        end_time = time.time()
+        last_start_time = start_time
+
+        slices = []
+        while last_start_time < end_time:
+            slices.append(
+                {
+                    "StartTime": last_start_time,
+                    # decrement second as API include records with specified StartTime and EndTime
+                    "EndTime": last_start_time + self.day_in_seconds - 1,
```

Comment:
WDYT about using a library like `pendulum` to do date math to avoid hardcoding constants? 
it would let us do something like
```
last_start_time.add(days=1).int_timestamp
```